libr/cons/cons.c

```diff
@@ -2146,6 +2146,42 @@ R_API void r_cons_cmd_help(const char *help[], bool use_color) {
 	}
 }
 
+/* See r_cons_cmd_help().
+ * This version will only print help for a specific command.
+ * Will append spec to cmd before looking for a match, if spec != 0.
+ */
+R_API void r_cons_cmd_help_match(const char *help[], bool use_color, R_BORROW R_NONNULL char *cmd, char spec) {
+	const char **match = NULL;
+	const char *match_help_text[4];
+	int i;
+
+	if (spec) {
+		/* We now own cmd */
+		cmd = r_str_newf ("%s%c", cmd, spec);
+	}
+
+	for (i = 0; help[i]; i += 3) {
+		if (!strcmp (help[i], cmd)) {
+			match = &help[i];
```

Comment:
only one match? whaht about strstr instead of strcmp? or add an option flag to specify that? i think it can be useful to filter the subcommands not just the specific command

Reply:
true, would be good to get both `pdr` and `pdr.` when you give it ("pd", 'r'). can add a `bool exact`.

could work if we return void, but gets more complicated if we want to return a pointer. maybe return a list?

Reply:
though then the caller would need to handle the new memory, im not sure about it. i dont think we can do both of these things

Reply:
Ok